app/src/main/java/com/github/se/eduverse/ui/quizz/QuizzScreen.kt

```diff
@@ -128,10 +141,22 @@ fun QuizScreen(navigationActions: NavigationActions, quizzRepository: QuizzRepos
                     Text("Generate Quiz", color = MaterialTheme.colorScheme.onPrimary)
                   }
 
```

Comment:
Currently, isLoading is set to true when the query to retrieve the questions begins, and false in the finally block. This ensures that the loading animation stops after the query, even if an error occurs. However, there is a potential problem if an exception occurs before isLoading = true is reached, or if multiple concurrent asynchronous actions modify isLoading. This could cause potential problems:

Visual inconsistency: If an exception occurs before isLoading = true is set, the loading animation does not appear and the user is unaware that something is happening.
Concurrency: If multiple calls to the onClick function are made in parallel (e.g., quickly clicking "Generate Quiz"), the isLoading state may be set incorrectly, resulting in a persistent or missing animation.